lib/ansiblelint/rules/PackageHasRetryRule.py

```diff
@@ -67,6 +67,16 @@ class PackageHasRetryRule(AnsibleLintRule):
         "zypper",
     ]
 
+    _module_ignore_states = [
+        "absent",
+    ]
+
     def matchtask(self, file, task):
-        return (task['action']['__ansible_module__'] in self.package_modules
-                and 'until' not in task)
+        module = task["action"]["__ansible_module__"]
+        if module in self._package_modules:
+            if 'until' not in task:
+                # Exclude tasks where state is `absent`.
+                if task['action'].get('state') not in self._module_ignore_states:
```

Comment:
This will collapse to
```suggestion
        return task['action'].get('state') not in self._module_ignore_states
```
after the suggestions above.